spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/SqlParametersFactory.java

```diff
@@ -0,0 +1,289 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jdbc.core.convert;
+
+import static org.springframework.data.jdbc.core.convert.SqlGenerator.*;
+
+import java.sql.SQLType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import org.springframework.data.jdbc.core.mapping.JdbcValue;
+import org.springframework.data.jdbc.support.JdbcUtil;
+import org.springframework.data.mapping.PersistentProperty;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.relational.core.dialect.Dialect;
+import org.springframework.data.relational.core.mapping.RelationalMappingContext;
+import org.springframework.data.relational.core.mapping.RelationalPersistentEntity;
+import org.springframework.data.relational.core.mapping.RelationalPersistentProperty;
+import org.springframework.data.relational.core.sql.IdentifierProcessing;
+import org.springframework.data.relational.core.sql.Select;
+import org.springframework.data.relational.core.sql.SqlIdentifier;
+import org.springframework.jdbc.support.JdbcUtils;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * Creates the {@link SqlIdentifierParameterSource} for various SQL operations, dialect identifier processing rules and
+ * applicable converters.
+ * 
+ * @author Jens Schauder
+ * @author Chirag Tailor
+ */
+public class SqlParametersFactory {
+	private final RelationalMappingContext context;
+	private final JdbcConverter converter;
+	private final Dialect dialect;
+
+	public SqlParametersFactory(RelationalMappingContext context, JdbcConverter converter, Dialect dialect) {
+		this.context = context;
+		this.converter = converter;
+		this.dialect = dialect;
+	}
+
+	/**
+	 * Creates the parameters for a SQL insert operation.
+	 *
+	 * @param instance the entity to be inserted. Must not be {@code null}.
+	 * @param domainType the type of the instance. Must not be {@code null}.
+	 * @param identifier information about data that needs to be considered for the insert but which is not part of the
+	 *          entity. Namely references back to a parent entity and key/index columns for entities that are stored in a
+	 *          {@link Map} or {@link List}.
+	 * @param includeId an indicator of whether the insert includes the id.
+	 * @return the {@link SqlIdentifierParameterSource} for the insert. Guaranteed to not be {@code null}.
+	 */
+	<T> SqlIdentifierParameterSource getInsert(T instance, Class<T> domainType, Identifier identifier, boolean includeId) {
```

Comment:
I find the name a little misleading, since the method does not returns "a Insert" which I would expect to be a SQL statement, or maybe an object wrapping such a statement. 

`forInsert` or `getParameterSourceForInsert` would be better names, I think.